core/domain/exp_services.py

```diff
@@ -758,6 +806,52 @@ def apply_change_list(exploration_id, change_list):
                         state_domain.WrittenTranslations.from_dict(
                             change.new_value))
                     state.update_written_translations(written_translations)
+                elif (
+                        change.property_name ==
+                        exp_domain.STATE_PROPERTY_IMAGE_ASSETS):
+
+                    if change.action == exp_domain.IMAGE_ASSETS_ADD_IMAGE:
+                        # Increment image_counter
+                        image_counter = exploration.image_counter
+                        image_counter += 1
```

Comment:
Is this always going to increase?

Reply:
Whenever there is change object for ``` add_image ```, then it always going to increase.